build-vm

```diff
@@ -734,6 +688,69 @@ vm_setup() {
     fi
 }
 
+#
+# create file system, mount file system to $BUILD_ROOT
+#
+vm_setup() {
+    vm_set_filesystem_type
+    vm_set_mount_options
+
+    vm_attach_root
+    setup_swap
+    echo "VM_ROOT: $VM_ROOT"
+    # this should not be needed, but sometimes a xen instance got lost
+    test "$VM_TYPE" = xen && vm_purge_xen
+    if test -n "$VMDISK_CLEAN" ; then
+	# delete old root to get rid of the old blocks
+	if test -n "$VM_ROOT" -a "$VM_ROOT_TYPE" = file -a -f "$VM_ROOT" ; then
+	    echo "Deleting old $VM_ROOT"
+	    rm -f "$VM_ROOT"
+	fi
+    fi
+    # using virt-make-fs for ext3 file VMDISK_FILESYSTEM
+    # so nothing is needed to do aprt from creating appropriate folders
+    # and calling virt-make-fs accordingly
+    if test ! "$USE_VIRT_FS" ; then
```

Comment:
Hmm, as the `if` statement basically goes to the end of the function, how about doing
```
if test -n "$USE_VIRT_FS" ; then
    return 0
fi
```

This should make the commit much smaller and less error prone. For example, you've moved the mkdir_build_root line in your commit by mistake.